lib/Lift/InstructionTranslator.cpp

```diff
@@ -38,438 +39,339 @@ static cl::opt<bool> RecordASM("record-asm",
 
 using IT = InstructionTranslator;
 
-namespace PTC {
-
-template<bool C>
-class InstructionImpl;
-
-enum ArgumentType {
-  In,
-  Out,
-  Const
-};
-
-template<typename T, typename Q, bool B>
-using RAI = RandomAccessIterator<T, Q, B>;
-
-template<ArgumentType Type, bool IsCall>
-class InstructionArgumentsIterator
-  : public RAI<uint64_t, InstructionArgumentsIterator<Type, IsCall>, false> {
-
-public:
-  using base = RandomAccessIterator<uint64_t,
-                                    InstructionArgumentsIterator,
-                                    false>;
-
-  InstructionArgumentsIterator &
-  operator=(const InstructionArgumentsIterator &R) {
-    base::operator=(R);
-    TheInstruction = R.TheInstruction;
-    return *this;
-  }
-
-  InstructionArgumentsIterator(const InstructionArgumentsIterator &R) :
-    base(R), TheInstruction(R.TheInstruction) {}
-
-  InstructionArgumentsIterator(const InstructionArgumentsIterator &R,
-                               unsigned Index) :
-    base(Index), TheInstruction(R.TheInstruction) {}
-
-  InstructionArgumentsIterator(PTCInstruction *TheInstruction, unsigned Index) :
-    base(Index), TheInstruction(TheInstruction) {}
-
-  bool isCompatible(const InstructionArgumentsIterator &R) const {
-    return TheInstruction == R.TheInstruction;
-  }
-
-public:
-  uint64_t get(unsigned Index) const;
-
-private:
-  PTCInstruction *TheInstruction;
-};
-
-template<>
-inline uint64_t
-InstructionArgumentsIterator<In, true>::get(unsigned Index) const {
-  return ptc_call_instruction_in_arg(&ptc, TheInstruction, Index);
+static uint64_t pc(LibTcgInstruction *Instr) {
+  revng_assert(Instr->opcode == LIBTCG_op_insn_start);
+  uint64_t PC = Instr->constant_args[0].constant;
+  if (Instr->nb_cargs > 1)
+    PC |= Instr->constant_args[1].constant << 32;
```

Comment:
Can you elaborate on why we need this?

Reply:
IIRC we are not forcing libtcg to an x64 host, and if you're lifting 64-bit code on a 32-bit host, the PC would be split into two registers for the `op_insn_start` instruction.

But we should force libtcg to lift as if it was running on a x64 host no matter what, for consistency of IR. I'll take a look at this.